tests/tests/test_regressions.py

```diff
@@ -25,56 +29,44 @@ def schedule_with_payload(base_url, payload, accept_400):
     return r.json()["ids"][0]
 
 
-def stdcm_with_payload(base_url, payload):
+def _stdcm_with_payload(base_url, payload):
     r = requests.post(base_url + "stdcm/", json=payload)
     if r.status_code // 100 != 2:
         if r.status_code // 100 == 4:
             return None
         raise RuntimeError(f"stdcm error {r.status_code}: {r.content}")
 
 
-def reproduce_test(path_to_json, *args, **kwargs):
-    base_url = kwargs["url"]
-    all_scenarios = kwargs["all_scenarios"]
+def _reproduce_test(path_to_json, all_scenarios):
     fuzzer_output = json.loads(path_to_json.read_bytes())
 
     if fuzzer_output["error_type"] == "STDCM":
-        stdcm_with_payload(base_url, fuzzer_output["stdcm_payload"])
+        _stdcm_with_payload(API_URL, fuzzer_output["stdcm_payload"])
         return True, ""
 
     stop_after_pathfinding = fuzzer_output["error_type"] == "PATHFINDING"
     stop_after_schedule = fuzzer_output["error_type"] == "SCHEDULE"
 
     scenario = all_scenarios[fuzzer_output["infra_name"]]
     timetable = scenario.timetable
-    path_id = pathfinding_with_payload(
-        base_url, fuzzer_output["path_payload"], scenario.infra, stop_after_pathfinding
-    )
+    path_id = _pathfinding_with_payload(API_URL, fuzzer_output["path_payload"], scenario.infra, stop_after_pathfinding)
     if stop_after_pathfinding:
         return True, ""
-    rolling_stock_id = get_rolling_stock(base_url)
+    rolling_stock_id = _get_rolling_stock_id(API_URL, "fast_rolling_stock")
 
     payload = fuzzer_output["schedule_payload"]
     payload["path"] = path_id
     payload["timetable"] = timetable
     payload["schedules"][0]["rolling_stock"] = rolling_stock_id
-    schedule_id = schedule_with_payload(base_url, payload, stop_after_schedule)
+    schedule_id = _schedule_with_payload(API_URL, payload, stop_after_schedule)
     if stop_after_schedule:
         return True, ""
 
-    r = requests.get(f"{base_url}train_schedule/{schedule_id}/result/")
+    r = requests.get(f"{API_URL}train_schedule/{schedule_id}/result/")
     if r.status_code // 100 != 2:
-        raise RuntimeError(
-            f"Schedule error {r.status_code}: {r.content}, id={schedule_id}"
-        )
-    return True, ""
-
-
-def list_tests():
-    dir = Path(__file__).parent.resolve()
-    for f in dir.glob("*.json"):
+        raise RuntimeError(f"Schedule error {r.status_code}: {r.content}, id={schedule_id}")
 
-        def run_test(func=f, *args, **kwargs):
-            return reproduce_test(func, *args, **kwargs)
 
-        yield run_test, f.stem
+@pytest.mark.parametrize("json_path", REGRESSION_TEST_JSON)
```

Comment:
Because the values are Path instances, the test names are `test_regressions[json_path0]`, this makes it harder to identify which test fails and to run a test individually.

If the values were strings, they'd be directly in the `[]`. I think the ideal behavior here would be to have the test parametrized with file name only as a string (or even .stem), and we look for the actual file inside the test.

Reply:
I have made modifications to have a cleaner name :
![image](https://user-images.githubusercontent.com/32448554/223985475-6abd8c3e-a12f-48bf-afd9-ded8b01cc33e.png)